libraries/AP_HAL_Linux/Util.cpp

```diff
@@ -183,24 +183,22 @@ int Util::get_hw_arm32()
     f = fopen("/proc/cpuinfo", "r");
 
     if (f == NULL) {
-        ret = -errno;
-        goto end;
+        return -errno;
     }
 
     while (fgets(buffer, MAX_SIZE_LINE, f) != NULL) {
         flag = strstr(buffer, hardware_description_entry);
-        if (flag != NULL) {
-            for (uint8_t i = 0; i < UTIL_NUM_HARDWARES; i++) {
-                if (strstr(buffer, _hw_names[i]) != 0) {
-                     ret = i;
-                     goto close_end;
-                }
+        if(flag == NULL) {
+            continue;
+        }
+        for (uint8_t i = 0; i < UTIL_NUM_HARDWARES; i++) {
+            if(strstr(buffer, _hw_names[i]) == 0) {
+                continue;
             }
+            fclose(f);
+            return i;
         }
     }
 
-close_end:
```

Comment:
Single exit point serves to avoid the exact problem this change includes: it leaks `f` if string is not found.